integrations/rudder-integration-appsflyer-react-native/ios/AppsFlyerAttribution.m

```diff
@@ -0,0 +1,100 @@
+//
+//  AppsFlyerAttribution.m
+//  react-native-appsflyer
+//
+//  Created by Amit Kremer on 11/02/2021.
+//
+
+#import <Foundation/Foundation.h>
+#import "AppsFlyerAttribution.h"
+
+@interface AppsFlyerAttribution ()
+@property NSUserActivity * userActivity;
+@property NSURL * url;
+@property NSString * sourceApplication;
+@property NSDictionary * options;
+
+@end
+
+@implementation AppsFlyerAttribution
+
++ (instancetype)shared {
+    static AppsFlyerAttribution *shared = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        shared = [[self alloc] init];
+    });
+    return shared;
+}
+
+- (id)init {
+    if (self = [super init]) {
+        _url = nil;
+        _userActivity = nil;
+        _sourceApplication = nil;
+        _options = nil;
+        _RNAFBridgeReady = NO;
+
+        [[NSNotificationCenter defaultCenter] addObserver:self
+                                                 selector:@selector(receiveBridgeReadyNotification:)
+                                                     name:RNAFBridgeInitializedNotification
+                                                   object:nil];
+    }
+    return self;
+}
+
+#pragma mark - AppDelegate methods
+
+- (void)continueUserActivity:(NSUserActivity *)userActivity restorationHandler:(void (^_Nullable)(NSArray * _Nullable))restorationHandler{
+    if([self RNAFBridgeReady] == YES){
+        [[AppsFlyerLib shared] continueUserActivity:userActivity restorationHandler:restorationHandler];
+    }else{
+        [self setUserActivity:userActivity];
+    }
+}
+
+- (void)handleOpenUrl:(NSURL *)url options:(NSDictionary *)options{
+    if([self RNAFBridgeReady] == YES){
+        [[AppsFlyerLib shared] handleOpenUrl:url options:options];
+    }else{
+        [self setUrl:url];
+        [self setOptions:options];
+    }
+}
+
+- (void)handleOpenUrl:(NSURL *)url sourceApplication:(NSString*)sourceApplication annotation:(id)annotation{
+    if([self RNAFBridgeReady] == YES){
+        [[AppsFlyerLib shared] handleOpenURL:url sourceApplication:sourceApplication withAnnotation:annotation];
+    }else{
+        [self setUrl:url];
+        [self setSourceApplication:sourceApplication];
+    }
+}
+
+#pragma mark - Bridge initialized notification
+
+- (void)receiveBridgeReadyNotification:(NSNotification *)notification {
+    // RD-69546
+    // start - We added this code because sometimes the SDK automatically resolves deeplinks on `applicationDidFinishLaunching`, and then when calling `continueUserActivity` on the same deeplink
+    // it skips them.
+    id AppsFlyer = [AppsFlyerLib shared];
```

Comment:
Why not `AppsFlyerLib` as `type`?

Reply:
Yes, we can define type as `AppsFlyerLib` in this case, but this part of the code, we directly took reference from Appsflyer RN SDK.

Reply:
Okay. Leave it for now.